server/src/main/java/org/elasticsearch/search/profile/SearchProfileQueryPhaseResults.java

```diff
@@ -0,0 +1,123 @@
+package org.elasticsearch.search.profile;
+
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.search.SearchPhaseResult;
+import org.elasticsearch.search.fetch.FetchSearchResult;
+import org.elasticsearch.search.profile.aggregation.AggregationProfileShardResult;
+import org.elasticsearch.search.profile.aggregation.AggregationProfiler;
+import org.elasticsearch.search.profile.query.QueryProfileShardResult;
+import org.elasticsearch.search.profile.query.QueryProfiler;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Profile results for the query phase run on all shards.
+ */
+public final class SearchProfileQueryPhaseResults implements Writeable { // This is Writeable for backwards compatibility
```

Comment:
For me it'd be clearer to call this `SearchProfileResultsBuilder` and rename `merge` -> `build`. It took me a while to wrap my head around all the classes.

Reply:
I hadn't thought of that name! I'll give it more thought because I'm kind of used to the old name. I understand wanting to change it, yeah.